src/app/course/components/btn-delete/btn-delete.component.ts

```diff
@@ -0,0 +1,31 @@
+import {Component, Input, OnInit} from '@angular/core';
```

Comment:
Usually such actions should be done in a parent component, not in buttons. It looks a bit strange and it's not good from performance side, cause you duplicate this functionality over many buttons instead of having it only one splace.